app/src/main/java/com/navapp/navigation/route/RouteLinkFactory.java

```diff
@@ -0,0 +1,68 @@
+package com.navapp.navigation.route;
+
+import android.net.Uri;
+
+import com.navapp.navigation.destination.Destination;
+
+public class RouteLinkFactory {
+    private static final String NAVIGATION_LINK =  "https://www.google.com/maps/dir/?api=1";
+    private static final String ORIGIN = "&origin=";
+    private static final String ORIGIN_PLACE_ID = "&origin_place_id=";
+    private static final String DESTINATION = "&destination=";
+    private static final String DESTINATION_PLACE_ID = "&destination_place_id=";
+    private static final String WAYPOINTS = "&waypoints=";
+    private static final String WAYPOINT_PLACE_ID = "&waypoint_place_ids=";
+    private static final char WAYPOINT_SEPARATOR = '|';
+
+    private static final String DRIVING_TRAVEL_MODE = "&travelmode=driving";
+    private static final String SKIP_OVERVIEW = "&dir_action=navigate";
+
+    public static Uri create(Route route, boolean chooseDriving, boolean shouldSkipOverview) {
+        String unEncodedUri = new StringBuilder()
+                .append(NAVIGATION_LINK)
+                .append(originSection(route))
+                .append(destinationSection(route))
+                .append(waypointsSection(route))
+                .append(chooseDriving ? DRIVING_TRAVEL_MODE : "")
+                .append(shouldSkipOverview ? SKIP_OVERVIEW : "")
+                .toString();
+
+        return Uri.parse(unEncodedUri);
+    }
+
+    private static String originSection(Route route) {
+        return new StringBuilder()
```

Comment:
It would be better to use the same `StringBuilder` as from the original caller instead.

Reply:
yeah ok